lib/adagrams.rb

```diff
@@ -0,0 +1,107 @@
+# Accesses the Dictionary file to read in user input later
+require 'csv'
+DICTIONARY = CSV.read('assets/dictionary-english.csv')
+
+LETTERS = {A: 9, B: 2, C: 2, D: 4, E: 12, F: 2, G: 3, H: 2, I: 9, J: 1, K: 1, L: 4, M: 2, N: 6, O: 8, P: 2, Q: 1, R: 6, S: 4, T: 6, U: 4, V: 2, W: 2, X: 1, Y: 2, Z: 1}
+
+# Method draws letters by plucking 10 random letters from the hash LETTERS and pushing it into a new array. 
+def draw_letters
+  letters_arr = []
+  # letters_arr contains every instance of every letter available in the tile bag (ex. there are 9 string "i"s in the letters_arr)
+  
+  index_of_letters = []
+  # index_of_letters contains the indexes of each instance of every letter in letters_arr
+  
+  letters_in_hand = []
+  # letters_in_hand contains the randomly selected letters from letters_arr
+  
+  LETTERS.each do |letter, qty|
+    qty.times do 
+      letters_arr.push(letter.to_s)
+    end
+  end
+  # We delete the index from index_of_letters in order to prevent pulling the same tile twice.
+  index_of_letters = (0..(letters_arr.length - 1)).to_a
+  pluck = 10
+  pluck.times do 
+    index = rand(0..(index_of_letters.length-1))
+    while !index_of_letters.include?(index)
+      index = rand(0..(index_of_letters.length-1))
+    end
+    letters_in_hand.push(letters_arr[index])
+    index_of_letters.delete(index)
+  end
+  return letters_in_hand
+end
+
+# This method checks that the letters_in_hand contains the plucked letters from previous method
+def uses_available_letters? (input, letters_in_hand)
```

Comment:
Minor nitpick: You CAN define a method with a space between the method name and the input parens, but Ruby will give a warning, because it prefers it if the method signature looks more like `def uses_available_letters?(input, letters_in_hand)`